### DISCUSSION TITLE
Function for Diesel Read operations (CRUD) of generic types

### DISCUSSION THREAD

**Discussion by schalimidi** [diesel-rs/diesel#4391] [Q&A]
The following function is for insert or create:

use diesel::query_builder::{InsertStatement};
use diesel::query_dsl::methods::{ExecuteDsl};

pub fn insert_into_table<T, M>(conn: &Pgconnection, table: T, records: M)
where
T: Table,
M: diesel::Insertable<T>,
InsertStatement<T, M::Values>: ExecuteDsl<PgConnection>,
{
diesel::insert_into(table)
.values(records)
.execute(conn);
}

Can do something like this for reads or load?

pub fn read_from_table<T, M>(conn: &Pgconnection, table: T, records: M)
where
T: Table,
SelectStatement<T>: ExecuteDsl<PgConnection>,
{
diesel::load(table)
.execute(conn);
}

**Comment by weiznich** [comment]
Its possible to write such generic functions but I advice not to do that in application code as you get a lot of complexity from the trait bounds and you essentially only replace a minor bit of code with such calls. In the end `insert_into_table(conn, users::table, values)` is not really shorter than `insert_into(users::table).values(values).execute(conn)`.

**Reply by schalimidi** [reply]
I have a situation where I have to load data from partition tables and the table name changes each time ( Ex: project_1_items, project_2_items etc). How to load from table partitions.

**Reply by weiznich** [reply]
partition tables are something that has no special support in diesel yet. If you want to use the built-in DSL you likely want to treat each partition as separate table and duplicate the relevant queries. One alternative might be to look into `diesel-dynamic-schema` which removes some restrictions around what needs to be statically known.

That written: I wrote above that this is not advised but nobody stops you implementing that. I just won't help with that as  previous experience with similar requests have shown that you will then end up with code that is likely not maintainable for you.
